node/test/client/src/block_builder.rs

```diff
@@ -88,7 +87,7 @@ impl InitPolkadotBlockBuilder for Client {
 		};
 
 		let mut block_builder = self
-			.new_block_at(&at, digest, false)
+			.new_block_at(&BlockId::Hash(hash), digest, false)
```

Comment:
I thought you're supposed to pass hash by value, no?

Reply:
Just following the function definition [here](https://github.com/paritytech/substrate/blob/master/client/block-builder/src/lib.rs#L121).

Yes `Hash` and also `BlockId` shall be passed by value, but obviously it is not always the case across substrate modules. Whenever changing the `BlockId` to `Hash` I was introducing by-value API.